### ISSUE TITLE
[WCAG 2.2 Draft Feedback] Success Criterion 2.4.12 Focus Not Obscured (Level AA)

### ISSUE THREAD

**Issue by dshoukry** [w3c/wcag#2710]
“Success Criterion 2.4.12 Focus Not Obscured (Minimum) (Level AA): When a user interface component receives keyboard focus, the component is not entirely hidden due to author-created content.”

Recommendation:
We would like the text to provide more examples and techniques, would like to understand implications to non-modal dialogs, overlaying content, and any amount of content being obscured, and have a few other testing and applicability questions. Finally, the examples, techniques, and failure seem to be incomplete - will there be another review for this newly-added SC or will this be added before Recommendation?

Please find detailed specifics covered in our [Sep22 2.4.12 Focus Not Obscured (AA) Google Doc](https://docs.google.com/document/d/1horL4-Nd4Ba1DNrfsFYnJqDD5-zO_XYzcFb-CYBeBEo/edit#)

**Comment by aardrian** [comment]
Can you please copy the entirety of your Google Doc into your issue here (and each other issue you filed today)?

At the very least, understand that the accessibility of Google Docs is not ideal. Asking readers to switch back and forth is at least a frustration to participation in the discussion and at most a barrier. Never mind the genuine concern that the document will not be maintained in perpetuity and not edited throughout this discussion.

**Comment by alastc** [comment]
Response for the group to review (with context):

<hr>

Regarding "When a user interface component receives keyboard focus":

> The timing element of this may contain some loopholes, but seems to be covered by WCAG 1.4.13, content on hover or focus.

Potentially. The main problem it was trying to solve was sticky headers/footers for which the key bit is when the focus lands on the control.

> If the focus indicator becomes obscured by a component that launches on a timer (such as a session timeout message or a chatbot message), will it still fail this criteria?
> 
> The criterion says that it applies “when a user interface component receives keyboard focus”, so we assume this scenario would not cause a failure, as long as the indicator was visible when it received focus.

I think that is a correct interpretation. Depending on the implementation, it might fail focus-order if the focus is not moved to the dialogue.

> Should this criterion be tested across all possible viewport sizes? If a focus indicator is obscured on a page only at a specific viewport size, should it be considered a failure? Does this also apply when the browser is zoomed (such as in 1.4.10 Reflow)?

Yes, the general approach in WCAG is that quite a few criteria need testing across "page variations". So you would test for each break point.

> Should this criteria be tested only with a keyboard or “keyboard like device”, and not with a mouse pointer? For example, if a tester uses their mouse to hover over an interactive element to display a non-modal dialog, then tabs to an element that is obscured by the dialog. Is this a valid way to test? Or should testers avoid using the mouse for this criterion?

This SC is focused on keyboard usage, so avoid using a mouse when testing. 

> If a focused element is positioned outside of the viewport (such as a “Skip to content” link that does not become visible on focus), would it be considered a failure of this criteria?

No, that isn't being hidden by author created content. However, it would fail focus-visible and probably focus-order as well.

**Comment by alastc** [comment]
The group approved the response above: 
https://www.w3.org/2023/02/07-ag-minutes#item10

**Comment by dshoukry** [comment]
> > If the focus indicator becomes obscured by a component that launches on a timer (such as a session timeout message or a chatbot message), will it still fail this criteria?
> > The criterion says that it applies “when a user interface component receives keyboard focus”, so we assume this scenario would not cause a failure, as long as the indicator was visible when it received focus.
> 
>>> I think that is a correct interpretation. Depending on the implementation, it might fail focus-order if the focus is not moved to the dialogue.

The person who shared this feedback wanted to note and confirm that: The challenging thing is that if there is a chatbot (or other nonmodal dialog that doesn't steal focus) dialog that appears 15 seconds after the page loads, it isn't a failure if the user focuses on the element before the dialog appears, but it is a failure if the dialog appears before the user navigates to the component. Would this be a failure if an element is introduced to the page dynamically that can obscure focus?
